cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -534,6 +546,110 @@ func (p *BicepProvider) purgeKeyVaults(
 	return nil
 }
 
+func (p *BicepProvider) getAppConfigsToPurge(
+	ctx context.Context,
+	groupedResources map[string][]azcli.AzCliResource,
+) ([]*azcli.AzCliAppConfig, error) {
+	configs := []*azcli.AzCliAppConfig{}
+
+	for resourceGroup, groupResources := range groupedResources {
+		for _, resource := range groupResources {
+			if resource.Type == string(infra.AzureResourceTypeAppConfig) {
+				config, err := p.azCli.GetAppConfig(ctx, p.env.GetSubscriptionId(), resourceGroup, resource.Name)
+				if err != nil {
+					return nil, fmt.Errorf("listing app configuration %s properties: %w", resource.Name, err)
+				}
+
+				if !config.Properties.EnablePurgeProtection {
+					configs = append(configs, config)
+				}
+			}
+		}
+	}
+
+	return configs, nil
+}
+
+// Azure AppConfigurations have a "soft delete" functionality (now enabled by default) where a configuration store may be marked
+// such that when it is deleted it can be recovered for a period of time. During that time, the name may not be reused.
+//
+// This means that running `az dev provision`, then `az dev infra delete` and finally `az dev provision`
+// again would lead to a deployment error since the configuration name is in use.
+//
+// Since that's behavior we'd like to support, we run a purge operation for each AppConfiguration after it has been deleted.
+//
+// See https://learn.microsoft.com/en-us/azure/azure-app-configuration/concept-soft-delete for more information on this feature.
+func (p *BicepProvider) purgeAppConfigs(
+	ctx context.Context,
+	asyncContext *async.InteractiveTaskContextWithProgress[*DestroyResult, *DestroyProgress],
+	appConfigs []*azcli.AzCliAppConfig,
+	options DestroyOptions,
+) error {
+	if len(appConfigs) > 0 && !options.Purge() {
+		appConfigWarning := fmt.Sprintf(""+
+			"\nThis operation will delete and purge %d App Configurations. "+
+			"These App Configurations have soft delete enabled allowing them to be recovered for a period \n"+
+			"of time after deletion. During this period, their names may not be reused.\n"+
+			"You can use argument --purge to skip this confirmation.\n\n",
+			len(appConfigs))
+
+		p.console.Message(ctx, output.WithWarningFormat(appConfigWarning))
+
+		err := asyncContext.Interact(func() error {
+			purgeAppConfigs, err := p.console.Confirm(ctx, input.ConsoleOptions{
+				Message: fmt.Sprintf(
+					"Would you like to %s delete these App Configurations instead, allowing their names to be reused?",
+					output.WithErrorFormat("permanently"),
+				),
+				DefaultValue: false,
+			})
```

Comment:
Question about adding yet another prompt/question for the user.  Instead of asking them to purge keyvaults, then purge app configs... Consider grouping all purgable resources into a single prompt??

@puicchan @savannahostrowski  ??

Reply:
https://github.com/Azure/azure-dev/issues/248#issuecomment-1278131763 Replied here!